### DISCUSSION TITLE
`503: Forbidden - Blocking Cross Origin request` when trying to launch on mybinder.org

### DISCUSSION THREAD

**Discussion by ethman** [executablebooks/meta#146] [Q&A] [answered]
Hello,

Thanks for this cool project! I set up a jupyter book in my [repo](https://github.com/source-separation/tutorial) and I want to have mybinder.org links. I set it up according to those docs (I think), but when I select the mybinder.org button on, say, [this page](https://source-separation.github.io/tutorial/basics/phase.html) the binder builds for a few seconds and ultimately the page finalizes on a message that says `403 Forbidden: Blocking Cross Origin request from https://mybinder.org/v2/gh/source-separation/tutorial/master?urlpath=tree/book/basics/phase.md.`.  However, I do see that a binder was built [here](https://nbviewer.jupyter.org/github/source-separation/tutorial/tree/master/)

I'm not sure what's going wrong. However I do see the same issue when pressing the Binder button on the [QuantEcon book](https://executablebooks.github.io/quantecon-example/docs/index.html).

Any tips?

Thanks,
-Ethan

Quick edit: Now I'm seeing a `404 Not Found` [error](https://hub.gke.mybinder.org/user/source-separation-tutorial-iez3ozob/tree/basics/phase.md). Still unsure of what's up and any help would be greatly appreciated! :D

**Comment by ethman** [comment, marked as answer]
Bump on this. Any help would be appreciated.

**Reply by choldgraf** [reply]
none of the links in your question are valid, they all result in 404 not found...

**Reply by ethman** [reply]
D'oh! I absentmindedly made the repo private. Thank you for following up. The repo is public again now so those links should work.

**Reply by choldgraf** [reply]
can you try Binder again to see if you still get a 404 error now that they're public?

**Reply by ethman** [reply]
Yeah, I just rebuilt the one launched from from [this page](https://source-separation.github.io/tutorial/basics/phase.html) and [here's the result](https://notebooks.gesis.org/binder/jupyter/user/source-separation-tutorial-c8s466m3/tree/basics/phase.md).

**Reply by choldgraf** [reply]
note that you can't share binder links like that, if you share a binder link for your specific session (one that has `/user/` in it), then the other person will get a 403 error

In addition, looking at your binder link your book exists in a sub-folder of the repo which is why the link results in a 404. See the "path_to_book" config documented here: https://jupyterbook.org/interactive/launchbuttons.html

**Reply by ethman** [reply]
Thanks for the info. I updated the config with the path to the book. Now when I try to launch the book via the binder link I see the 503 error like I was seeing in the original post.

**Reply by choldgraf** [reply]
is this the link you cliked:

https://mybinder.org/v2/gh/source-separation/tutorial/master?urlpath=tree/book/basics/phase.md

? that is the only link that will work without resulting in a 403 error, you cannot share `/user/` links, and those links will stop working after 15 minutes of inactivity.

**Reply by ethman** [reply]
Yeah, when I click the link at the top of that jupyter book url, I get taken to the mybinder page, which spins for a bit (I can see the logs building/retrieving the notebook), and then all I see is the 403 error. I'm not sharing the page at all.

**Reply by ethman** [reply]
I have a few ad blockers that I'm running in firefox, could there be some conflict with those somehow?

**Reply by choldgraf** [reply]
hmm I just tried that link and got the same thing...that's pretty weird. I feel like this is _probably_ a binder bug as opposed to a Jupyter Book bug. I opened up an issue here to figure it out: https://github.com/jupyterhub/mybinder.org-deploy/issues/1638

**Reply by ethman** [reply]
Using ipynb will work perfectly fine. Thanks so much for your help!
